buildpack.go

```diff
@@ -0,0 +1,201 @@
+package libdependency
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Masterminds/semver/v3"
+	"github.com/joshuatcasey/collections"
+	"github.com/paketo-buildpacks/libdependency/versionology"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"golang.org/x/exp/slices"
+)
+
+func ParseBuildpackToml(buildpackTomlPath string) (cargo.Config, error) {
```

Comment:
Code comments would be super helpful here. Also are these user-facing functions? If so can we document them in the README? How would a buildpack author know which of these they need?

Reply:
If they're more of an internal package, let's move them into a new directory called `internal` for clarity